fuse/test/flock_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 the Go-FUSE Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package test
+
+import (
+	"os"
+	"os/exec"
+	"regexp"
+	"syscall"
+	"testing"
+)
+
+func TestFlock(t *testing.T) {
+	tc := NewTestCase(t)
+	defer tc.Cleanup()
+
+	contents := []byte{1, 2, 3}
+	tc.WriteFile(tc.origFile, []byte(contents), 0700)
+
+	f, err := os.OpenFile(tc.mountFile, os.O_WRONLY, 0)
+	if err != nil {
+		t.Fatalf("OpenFile(%q): %v", tc.mountFile, err)
+	}
+	defer f.Close()
+
+	if err = syscall.Flock(int(f.Fd()), syscall.LOCK_EX); err != nil {
+		t.Errorf("Flock returned: %v", err)
+		return
+	}
+
+	if out, err := runExternalFlock(tc.mountFile); err != nil {
```

Comment:
I'd rather use that, and skip the test if we can't find the binary. Running the test binary from within the test binary sounds like it might easily break